kernel/thread.c

```diff
@@ -908,7 +908,15 @@ static inline int z_vrfy_k_float_disable(struct k_thread *thread)
 #endif /* CONFIG_USERSPACE */
 
 #ifdef CONFIG_IRQ_OFFLOAD
+
+/* Make offload_sem visible outside under testing, in order to release
+ * it outside when error happened.
+ */
+#ifdef CONFIG_ZTEST
+K_SEM_DEFINE(offload_sem, 1, 1);
+#else
 static K_SEM_DEFINE(offload_sem, 1, 1);
```

Comment:
you can just delete the static keyword, no need for the ifdef.

I would accept a patch which makes the entire irq_offload() API CONFIG_ZTEST only, to make it clear that this is only for tests.

Reply:
OK, got it. And I think I would just remove the static. That means not only CONFIG_ZTEST but CONFIG_TEST could use it.  

And may I have you opinion again? I guess there might be few other kinds of resources, would encounter the problem like this during some negative(error) case. If this happened, do you think is it better to have a extra stand-alone CONFIG to get them all visible? such as CONFIG_NEGATIVE_TEST or CONFIG_TEST_ERROR_RECOVER? Thanks!

Reply:
I think CONFIG_TEST is enough.

Reply:
Got it, thanks!